web/src/compose_recipient.ts

```diff
@@ -231,8 +248,29 @@ function item_click_callback(event, dropdown) {
     event.stopPropagation();
 }
 
-function get_options_for_recipient_widget() {
-    const options = stream_data.get_options_for_dropdown_widget();
+function get_options_for_stream_dropdown_widget(): dropdown_widget.Option[] {
+    const streams = stream_data.subscribed_subs();
+    const direct_messages_option: DirectMessagesOption = {
+        is_direct_message: false,
+        unique_id: "",
+        name: "",
+    };
+
+    const options = [
+        direct_messages_option,
+        ...streams
+            .map((stream) => ({
+                name: stream.name,
+                unique_id: stream.stream_id,
+                stream,
+            }))
+            .sort((a, b) => util.strcmp(a.name.toLowerCase(), b.name.toLowerCase())),
+    ];
+    return options;
+}
+
```

Comment:
This seems to be adding a bunch of new code; I'm guessing you misresolved a merge conflict or something. What's the story with this?

Reply:
Oh, the function I added here doesn't have to do with a merge conflict. Actually, in the `get_options_for_recipient_widget()` method, the `get_options_for_dropdown_widget()` from `stream_data` module is called. This is the current implementation. However, this method, doesn't handle data attributes for Direct messages, which causes a type error. I thought to refactor this method in `stream_data`, but I thought not to because I reasoned it's best to keep the method the same since the module, as the name suggest, is meant to handle data related only to `stream`. 

Hence, I created a method internally for `compose_recipient` to handle both stream data and direct message attributes.